mempool/mempool.go

```diff
@@ -37,6 +38,14 @@ func (p *Pool) Enqueue(userTxn comettypes.Tx) error {
 	// Unfortunately, comet's DB interface doesn't support it.
 	// Moving to a different DB interface is left for future work.
 
+	// Attempt to adapt the Cosmos transaction to an Ethereum deposit transaction.
+	// If the adaptation results in one or more transactions, it indicates that the
+	// user transaction is a deposit transaction, which is not allowed in the pool.
+	if txs, _ := rolluptypes.AdaptCosmosDepositTxToEthTx(userTxn); len(txs) > 0 {
```

Comment:
No, it doesn't work that way. In the case of `err == nil` when I run `make test`
```
--- FAIL: TestBroadcastTx (0.01s)
    comet_test.go:128: 
                Error Trace:    /Users/daniilankusin/monomer/comet/comet_test.go:128
                Error:          Received unexpected error:
                                enqueue in mempool: deposit txs are not allowed in the pool
                Test:           TestBroadcastTx
FAIL
```

Reply:
Hmmm, why does the test fail?

Reply:
Because of the functionality of the adapter in [here](https://github.com/AnkushinDaniil/monomer/blob/fb07012e7740b192fe4d2a57293afc904d90c231/x/rollup/types/adapters.go#L113-L115). It calls break, not an error. If I change it to return an error, the behaviour of the function is false. I can implement another function, but I don't think it's necessary. It would just be a wrap or very similar function.

Reply:
We should change the `break` to return an error.

Reply:
Yes, I tried that. In case of an error instead of break we fail other tests:
```
--- FAIL: TestBuild (0.02s)
    --- FAIL: TestBuild/txs_in_inclusion_list (0.01s)
        builder_test.go:135: 
                Error Trace:    /Users/daniilankusin/monomer/builder/builder_test.go:135
                Error:          Received unexpected error:
                                make block: convert block to Ethereum representation: adapt txs: adapt cosmos deposit tx: unmarshal binary tx: transaction type not supported
                Test:           TestBuild/txs_in_inclusion_list
    --- FAIL: TestBuild/txs_in_mempool (0.00s)
        builder_test.go:91: 
                Error Trace:    /Users/daniilankusin/monomer/builder/builder_test.go:91
                Error:          Received unexpected error:
                                deposit txs are not allowed in the pool
                Test:           TestBuild/txs_in_mempool
    --- FAIL: TestBuild/txs_in_mempool_and_inclusion_list (0.00s)
        builder_test.go:91: 
                Error Trace:    /Users/daniilankusin/monomer/builder/builder_test.go:91
                Error:          Received unexpected error:
                                deposit txs are not allowed in the pool
                Test:           TestBuild/txs_in_mempool_and_inclusion_list
    --- FAIL: TestBuild/txs_in_mempool_and_inclusion_list_with_NoTxPool (0.00s)
        builder_test.go:91: 
                Error Trace:    /Users/daniilankusin/monomer/builder/builder_test.go:91
                Error:          Received unexpected error:
                                deposit txs are not allowed in the pool
                Test:           TestBuild/txs_in_mempool_and_inclusion_list_with_NoTxPool
--- FAIL: TestRollback (0.01s)
    builder_test.go:252: 
                Error Trace:    /Users/daniilankusin/monomer/builder/builder_test.go:252
                Error:          Received unexpected error:
                                make block: convert block to Ethereum representation: adapt txs: adapt cosmos deposit tx: unmarshal binary tx: transaction type not supported
                Test:           TestRollback
FAIL
FAIL    github.com/polymerdao/monomer/builder   1.042s
--- FAIL: TestBroadcastTx (0.00s)
    comet_test.go:128: 
                Error Trace:    /Users/daniilankusin/monomer/comet/comet_test.go:128
                Error:          Received unexpected error:
                                enqueue in mempool: deposit txs are not allowed in the pool
                Test:           TestBroadcastTx
```

Reply:
I will try to fix the builder tests because I support your point of view, I don't feel that we can leave `break` in there.

Reply:
We can fix the tests using the transaction helpers in `testutils` I think.

Reply:
Yes, I tried that, it results in this:
```
--- FAIL: TestBuild (0.03s)
    --- FAIL: TestBuild/txs_in_inclusion_list (0.00s)
        builder_test.go:135: 
                Error Trace:    /Users/daniilankusin/monomer/builder/builder_test.go:135
                Error:          Received unexpected error:
                                make block: convert block to Ethereum representation: adapt txs: adapt cosmos deposit tx: unmarshal binary tx: transaction type not supported
                Test:           TestBuild/txs_in_inclusion_list
    --- FAIL: TestBuild/txs_in_mempool (0.01s)
        builder_test.go:135: 
                Error Trace:    /Users/daniilankusin/monomer/builder/builder_test.go:135
                Error:          Received unexpected error:
                                make block: convert block to Ethereum representation: adapt txs: adapt cosmos deposit tx: unmarshal binary tx: transaction type not supported
                Test:           TestBuild/txs_in_mempool
    --- FAIL: TestBuild/txs_in_mempool_and_inclusion_list (0.00s)
        builder_test.go:135: 
                Error Trace:    /Users/daniilankusin/monomer/builder/builder_test.go:135
                Error:          Received unexpected error:
                                make block: convert block to Ethereum representation: adapt txs: adapt cosmos deposit tx: unmarshal binary tx: transaction type not supported
                Test:           TestBuild/txs_in_mempool_and_inclusion_list
    --- FAIL: TestBuild/txs_in_mempool_and_inclusion_list_with_NoTxPool (0.00s)
        builder_test.go:135: 
                Error Trace:    /Users/daniilankusin/monomer/builder/builder_test.go:135
                Error:          Received unexpected error:
                                make block: convert block to Ethereum representation: adapt txs: adapt cosmos deposit tx: unmarshal binary tx: transaction type not supported
                Test:           TestBuild/txs_in_mempool_and_inclusion_list_with_NoTxPool
--- FAIL: TestRollback (0.01s)
    builder_test.go:252: 
                Error Trace:    /Users/daniilankusin/monomer/builder/builder_test.go:252
                Error:          Received unexpected error:
                                make block: convert block to Ethereum representation: adapt txs: adapt cosmos deposit tx: unmarshal binary tx: transaction type not supported
                Test:           TestRollback
```
I support your point, I don't think we can leave `break` in there
I will try ti fix `builder_test.go`

Reply:
upd:
Used testutils to generate txs -> panic due to equality of txs
Fixed panic with randomly generated txs
Errors remain:
```
--- FAIL: TestBuild (0.04s)
    --- FAIL: TestBuild/txs_in_inclusion_list (0.01s)
        /Users/daniilankusin/monomer/builder/builder_test.go:133: 
            	Error Trace:	/Users/daniilankusin/monomer/builder/builder_test.go:133
            	Error:      	Received unexpected error:
            	            	make block: convert block to Ethereum representation: adapt txs: cosmos deposit: unmarshal cosmos tx: proto: illegal wireType 6
            	Test:       	TestBuild/txs_in_inclusion_list
    --- FAIL: TestBuild/txs_in_mempool (0.01s)
        /Users/daniilankusin/monomer/builder/builder_test.go:133: 
            	Error Trace:	/Users/daniilankusin/monomer/builder/builder_test.go:133
            	Error:      	Received unexpected error:
            	            	make block: convert block to Ethereum representation: adapt txs: cosmos deposit: unmarshal cosmos tx: proto: Tx: illegal tag 0 (wire type 2)
            	Test:       	TestBuild/txs_in_mempool
    --- FAIL: TestBuild/txs_in_mempool_and_inclusion_list (0.00s)
        /Users/daniilankusin/monomer/builder/builder_test.go:133: 
            	Error Trace:	/Users/daniilankusin/monomer/builder/builder_test.go:133
            	Error:      	Received unexpected error:
            	            	make block: convert block to Ethereum representation: adapt txs: cosmos deposit: unmarshal cosmos tx: proto: illegal wireType 6
            	Test:       	TestBuild/txs_in_mempool_and_inclusion_list
    --- FAIL: TestBuild/txs_in_mempool_and_inclusion_list_with_NoTxPool (0.01s)
        /Users/daniilankusin/monomer/builder/builder_test.go:133: 
            	Error Trace:	/Users/daniilankusin/monomer/builder/builder_test.go:133
            	Error:      	Received unexpected error:
            	            	make block: convert block to Ethereum representation: adapt txs: cosmos deposit: unmarshal cosmos tx: proto: illegal wireType 6
            	Test:       	TestBuild/txs_in_mempool_and_inclusion_list_with_NoTxPool
```